src/main/java/com/github/onsdigital/babbage/search/external/requests/base/AbstractSearchRequest.java

```diff
@@ -26,23 +29,29 @@ public AbstractSearchRequest(Class<T> returnClass) {
      * Abstract method for building/returning the target URI for HTTP requests
      * @return
      */
-    public abstract String targetUri();
+    public abstract URIBuilder targetUri();
 
     /**
      * Builds a simple HTTP GET request with the target URI
      * @return
      * @throws Exception
      */
     protected Request get() throws Exception {
-        return SearchClient.getInstance().get(this.targetUri());
+        if (searchClient == null) {
```

Comment:
minor point but consider adding a `getSearchClient()` method which does the null check / assignment. You will only be doing it in one place then.